core/src/test/resources/newlines/alwaysBeforeTopLevelStatements.stat

```diff
@@ -144,4 +158,117 @@ object A {
 
   @Singleton
   class B @Inject()(val x: Int)
+
+  @js.native
+  class C() {
+    def c = 5
+  }
+}
+<<< Double annotation
+object a {
+  @annot
+  @bannot
+  class A
+}
+>>>
+object a {
+
+  @annot
+  @bannot
+  class A
+}
+<<< Don't add NL before single-line defs
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+>>>
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+
+<<< Add NL before multi-line statements
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+  def foo = {
+    println("foo")
+  }
+  def bar = {
+    println("bar")
+  }
+}
+>>>
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+<<< nested defs (up for debate)
+object a {
+  def foo = {
+    def b = {
+      3
+    }
+    def c = {
+      4
+    }
+  }
+}
+>>>
+object a {
+
+  def foo = {
+
+    def b = {
```

Comment:
This new scheme of detecting top level statements is very easy to extend. The trick is to replace this `collect` here https://github.com/scalameta/scalafmt/pull/904/files#diff-4f68f079630ceb866ab2d90e4eacf3e3R182 with a custom `Traverser`. That will make it possible to for example exclude the first statement of `Template`.

We can leave this for a separate PR.